apps/io-wallet-user-func/src/infra/azure/storage/queue.ts

```diff
@@ -0,0 +1,43 @@
+import { QueueClient } from "@azure/storage-queue";
+import * as E from "fp-ts/lib/Either";
+import { stringify } from "fp-ts/lib/Json";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+
+class StorageQueueError extends Error {
+  errorCode?: string;
+  name = "StorageQueueError";
+  constructor(errorCode?: string) {
+    super("Unable to enqueue the message.");
+    this.errorCode = errorCode;
+  }
+}
+
+const toBase64 = flow(Buffer.from, (b) => b.toString("base64"));
+
+const sendMessage =
+  (
+    message: string,
+  ): RTE.ReaderTaskEither<
+    { queueClient: QueueClient },
+    StorageQueueError,
+    void
+  > =>
+  ({ queueClient }) =>
+    pipe(
+      TE.tryCatch(() => queueClient.sendMessage(message), E.toError),
+      TE.filterOrElse(
+        (response) => response.errorCode === undefined,
+        (response) => new StorageQueueError(response.errorCode),
+      ),
+      TE.map(() => undefined),
+    );
+
+export const enqueue = flow(
```

Comment:
What do you think about adding a check on what type of data enqueue should expect based on the queue type?